packages/block-library/src/post-excerpt/edit.js

```diff
@@ -2,16 +2,96 @@
  * WordPress dependencies
  */
 import { useEntityProp, useEntityId } from '@wordpress/core-data';
-import { PlainText } from '@wordpress/block-editor';
+import { useMemo } from '@wordpress/element';
+import { InspectorControls, RichText } from '@wordpress/block-editor';
+import { PanelBody, RangeControl, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
 
-function PostExcerptDisplay() {
+function usePostContentExcerpt( excerptLength ) {
+	const [ , , { raw: rawPostContent } ] = useEntityProp(
+		'postType',
+		'post',
+		'content'
+	);
+	return useMemo( () => {
+		if ( ! rawPostContent ) {
+			return '';
+		}
+		const excerptElement = document.createElement( 'div' );
+		excerptElement.innerHTML = rawPostContent;
+		const excerpt =
+			excerptElement.textContent || excerptElement.innerText || '';
+		return excerpt
+			.trim()
+			.split( ' ', excerptLength )
+			.join( ' ' );
+	}, [ rawPostContent, excerptLength ] );
+}
+
+function PostExcerptEditor( {
+	attributes: { excerptLength, moreText, showMoreOnNewLine },
+	setAttributes,
+	isSelected,
+} ) {
 	const [ excerpt, setExcerpt ] = useEntityProp( 'postType', 'post', 'excerpt' );
-	return <PlainText value={ excerpt } onChange={ setExcerpt } />;
+	const postContentExcerpt = usePostContentExcerpt( excerptLength );
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Post Excerpt Settings' ) }>
+					{ ! excerpt && (
+						<RangeControl
+							label={ __( 'Max words' ) }
+							value={ excerptLength }
+							onChange={ ( newExcerptLength ) =>
+								setAttributes( { excerptLength: newExcerptLength } )
+							}
+							min={ 10 }
+							max={ 100 }
+						/>
+					) }
+					<ToggleControl
+						label={ __( 'Show link on new line' ) }
+						checked={ showMoreOnNewLine }
+						onChange={ ( newShowMoreOnNewLine ) =>
+							setAttributes( { showMoreOnNewLine: newShowMoreOnNewLine } )
+						}
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<RichText
+				className={
+					! showMoreOnNewLine && 'wp-block-post-excerpt__excerpt is-inline'
+				}
+				placeholder={ postContentExcerpt }
+				value={ excerpt || ( isSelected ? '' : postContentExcerpt ) }
+				onChange={ setExcerpt }
+				keepPlaceholderOnFocus
+			/>
+			{ ! showMoreOnNewLine && ' ' }
+			<RichText
```

Comment:
Should this be inside a paragraph if "showMoreOnNewLine" is true (to match the backend)

Reply:
Yes, updated.